hail/python/hail/expr/expressions/base_expression.py

```diff
@@ -206,6 +206,15 @@ def to_expr(e, dtype=None) -> 'Expression':
     else:
         return hl.literal(x, dtype)
 
+def cast_expr(e, dtype=None) -> 'Expression':
```

Comment:
This is the just the non-typechecking portion of `to_expr`, right? You could probably rewrite `to_expr` in terms of this to avoid duplication logic, or implement this as a `coerce=False` flag on `to_expr`, or something.

In terms of the actual logic---I'm kind of curious what happens if you try to do
```
hl.eval(cast_expr({'a': 5, 'b': 5}, dtype='struct{a:float32, b:float64}'))
```
or something?

Reply:
Yeah, I'll rewrite to_expr in terms of this